web/components/input/buttons/CheckboxGroup/index.js

```diff
@@ -57,6 +57,20 @@ CheckboxGroup.propTypes = {
     PropTypes.element,
     PropTypes.array,
   ]),
+  // extra CSS class names for the container (not applied to children)
+  className: PropTypes.string,
+  /* Additional optional parameters implicitly passed down to children
+
+      // see Checkbox.propTypes
+      name,
+      value,
+      defaultChecked,
+      disabled,
+      onChange,
+
+      // passed down to all children (but not applied to container)
+      sharedClassName: PropTypes.string,
```

Comment:
shouldn't this be a proptype of checkbox?

Reply:
I'm not sure I understand how this works, or why it's needed - I can think of two situations where children are generated:

1. User provides children to class. In this case, sharedClassName is useless, because you can just attach className to the provided children
2. Component generates children. In this case, isn't sharedClassName also useless, because you don't want the user to influence how these children are generated, because they are tied to the parent component? Why can't the parent just provide the desired classNames?

can you provide an example/point us to one? a quick cmd-f doesn't find me any usage of this prop

Reply:
In `RadioGroup` and `CheckboxGroup`, `className` is intended to apply class names to that container, and so it's not passed down to the children (unlike `name`, `disabled`, `selected/defaultChecked`, and `onChange` props). However, `sharedClassName` is passed down to the children.

So instead of this...
```
<RadioGroup name="test_radio_group">
  <RadioButton label="Label 1" value="value1" className="margin-right-m" />
  <RadioButton label="Label 2" value="value2" className="margin-right-m" />
  <RadioButton label="Label 3" value="value3" className="margin-right-m" />
</RadioGroup>
```
... we can do this:
```
<RadioGroup name="test_radio_group" sharedClassName="margin-right-m">
  <RadioButton label="Label 1" value="value1" />
  <RadioButton label="Label 2" value="value2" />
  <RadioButton label="Label 3" value="value3" />
</RadioGroup>
```

This is also the reason why `sharedClassName` isn't a prop of the children components.

In retrospect, the convenience is probably not worth the confusion (if you define `className` for a `RadioButton`/`Checkbox`, it will override `sharedClassName` instead of adding to it). I can revert it back to the state where class names are not passed down to the children (similar to the `value` prop) and must be explicitly defined for each child. Let me know which one you guys like more.

Reply:
Yeah, I think it's probably best to be more explicit and let className be defined for each child. If you find it verbose, a simple `map` will do the trick:

```js
<RadioGroup name="test_radio_group">
  {radioButtons.map(b => (<RadioButton
    key={b.label} 
    label={b.label}
    value={b.value} 
    className="margin-right-m" />)}
</RadioGroup>
```